.github/workflows/ci.yml

```diff
@@ -20,29 +20,24 @@ jobs:
             otp: 20.3.8.26
           - elixir: 1.9.4
             otp: 20.3.8.26
+            warnings_as_errors: true # not 1.10 as its --warnigs-as-errors has bugs https://github.com/elixir-lang/elixir/issues/10073
           - elixir: 1.10.4
             otp: 21.3.8.16
           - elixir: 1.10.4
             otp: 23.0.3
+            check_formatted: true
     steps:
       - uses: actions/checkout@v2.3.1
       - uses: actions/setup-elixir@v1.3.0
         with:
           otp-version: ${{matrix.otp}}
           elixir-version: ${{matrix.elixir}}
+      - run: mix format --check-formatted
+        if: matrix.check_formatted
       - name: Install Dependencies
         run: |
           mix local.hex --force
           mix deps.get --only test
+      - run: mix compile --warnings-as-errors
```

Comment:
Could you change this to `MIX_ENV=test mix ...`? otherwise we’d unnecessarily compile the project again when running `mix test` below.

Reply:
Is it the case that `MIX_ENV=<env_name> mix compile --warnings-as-errors` may produce different results depending on `env_name`?

Reply:
The end result will most likely be the same for libraries but it will differ for apps since it is common to have test-specific configs in apps (but not in libs).

Reply:
> The end result will **most likely** be the same for libraries

In case they differ, what is the recommended course of action? Only care about `MIX_ENV=test`? Make sure `MIX_ENV=dev` is also succesful?

Reply:
It depends on your goals. If your goal is actually how it will behave as a dep, then MIX_ENV=prod would be the best, as deps always run in prod.

Reply:
So why test only on `MIX_ENV=test` on CI then?

Reply:
You asked what to do in case they differ. They don't differ in 99% of the cases, that's why only MIX_ENV=test is enough.